appservice/src/main/java/nu/yona/server/subscriptions/rest/UserPhotoController.java

```diff
@@ -0,0 +1,151 @@
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.imageio.ImageIO;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.ExposesResourceFor;
+import org.springframework.hateoas.Resource;
+import org.springframework.hateoas.mvc.ControllerLinkBuilder;
+import org.springframework.hateoas.mvc.ResourceAssemblerSupport;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+
+import nu.yona.server.crypto.seckey.CryptoSession;
+import nu.yona.server.exceptions.InvalidDataException;
+import nu.yona.server.exceptions.YonaException;
+import nu.yona.server.rest.Constants;
+import nu.yona.server.subscriptions.rest.UserPhotoController.UserPhotoResource;
+import nu.yona.server.subscriptions.service.UserPhotoDto;
+import nu.yona.server.subscriptions.service.UserPhotoService;
+import nu.yona.server.subscriptions.service.UserService;
+
+@Controller
+@ExposesResourceFor(UserPhotoResource.class)
+@RequestMapping(value = "/", produces = { MediaType.APPLICATION_JSON_VALUE })
+public class UserPhotoController
+{
+	@Autowired
+	private UserService userService;
+
+	@Autowired
+	private UserPhotoService userPhotoService;
+
+	@RequestMapping(value = "/userPhotos/{userPhotoId}", method = RequestMethod.GET, produces = { MediaType.IMAGE_PNG_VALUE })
+	@ResponseBody
+	public ResponseEntity<byte[]> getUserPhoto(@PathVariable UUID userPhotoId)
+	{
+		return new ResponseEntity<>(userPhotoService.getUserPhoto(userPhotoId).getPngBytes(), HttpStatus.OK);
+	}
+
+	@RequestMapping(value = "/users/{userId}/photo", method = RequestMethod.PUT, consumes = "multipart/form-data")
+	@ResponseBody
+	public ResponseEntity<UserPhotoResource> uploadUserPhoto(
+			@RequestHeader(value = Constants.PASSWORD_HEADER) Optional<String> password,
+			@RequestParam(value = "file", required = false) MultipartFile userPhoto, @PathVariable UUID userId)
+	{
+		try (CryptoSession cryptoSession = CryptoSession.start(password, () -> userService.canAccessPrivateData(userId)))
+		{
+			return createUserPhotoUploadResponse(
+					userPhotoService.addUserPhoto(userId, UserPhotoDto.createUnsavedInstance(getPngBytes(userPhoto))));
+		}
+	}
+
+	@RequestMapping(value = "/users/{userId}/photo", method = RequestMethod.DELETE)
+	@ResponseBody
+	public ResponseEntity<Void> removeUserPhoto(@RequestHeader(value = Constants.PASSWORD_HEADER) Optional<String> password,
```

Comment:
If you annotate this method with ``@ResponseStatus(HttpStatus.OK)``, you can change the return type to simply ``void`` and remove line 83.

Reply:
Will do